src/command.rs

```diff
@@ -1,13 +1,37 @@
+use std::path::Path;
+
 //Git Command
-use std::process::{Command, Output};
-use std::str;
-
-pub fn git_clone(proj_name: &str, url: String) -> Output {
-    let output = Command::new("git")
-        .arg("clone")
-        .arg(url)
-        .arg(proj_name)
-        .output()
-        .expect("failed to execute process");
-    output
+use git2::{build::RepoBuilder, Error, FetchOptions, RemoteCallbacks, Repository};
+use progress_bar::{
+    color::{Color, Style},
+    progress_bar::ProgressBar,
+};
+
+use crate::size::get_repo_size;
+
+pub fn git_clone(proj_name: &str, url: String) -> Result<Repository, Error> {
+    println!("PROJ NAME = {}", &proj_name);
+    let repo_size = get_repo_size(&url);
+
+    let mut progress_bar = ProgressBar::new(repo_size / 1000);
```

Comment:
this might vary and even make the repo size 0. Why not set a 0 to 100% and map the repo size to that range?